src/features/search/KennisartikelDetail.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <article>
+    <utrecht-heading model-value :level="level" class="heading">
+      {{ title }}
+    </utrecht-heading>
+    <nav>
+      <ul>
+        <li
+          v-for="({ isActive, label, id }, i) in mappedSections"
+          :key="id + 'nav'"
+          :class="{ 'is-active': isActive }"
+        >
+          <span v-if="isActive">{{ label }}</span>
+          <a v-else :href="`#${id}`" @click.prevent="currentSectionIndex = i">{{
+            label
+          }}</a>
+        </li>
+      </ul>
+    </nav>
+    <section
+      v-for="{ id, html, isActive, label } in mappedSections"
+      :key="id + 'text'"
+      :class="{ 'is-active': isActive }"
+      :id="id"
+    >
+      <utrecht-heading model-value :level="level + 1">{{
+        label
+      }}</utrecht-heading>
+      <div v-html="html"></div>
+    </section>
+  </article>
+</template>
+<script setup lang="ts">
+import { cleanHtml, unEscapeHtml } from "@/helpers/html";
+import { UtrechtHeading } from "@utrecht/web-component-library-vue";
+import { nanoid } from "nanoid";
+import { computed, ref, watch } from "vue";
+
+const language = "nl";
+const knownSections = {
+  specifiekeTekst: "Inleiding",
+  procedureBeschrijving: "Aanvraag",
+  conditions: "Voorwaarden",
+  bewijs: "Bewijs",
+  kostenEnBetaalmethoden: "Kosten",
+  uitersteTermijn: "Termijn",
+  bezwaarEnBeroep: "Bezwaar",
+  vereisten: "Vereisten",
+  notice: "Bijzonderheden",
+  wtdBijGeenReactie: "Geen reactie",
+  contact: "Contact",
+  deskMemo: "KCC",
+} as const;
+
+const uniqueId = nanoid();
+
+const props = defineProps<{
+  jsonObject: any;
+  title: string;
+  level: 2 | 3 | 4;
+}>();
+
+const currentSectionIndex = ref(0);
+
+const translation = computed(() =>
+  props.jsonObject.vertalingen.find((x: any) => x.taal === language)
+);
+
+const sections = computed(() => {
+  const translationObj = translation.value;
+  const level = props.level;
+  return Object.entries(knownSections)
+    .map(([key, label], i) => {
+      const text = translationObj[key];
+      if (!text) return undefined;
+      const unEscaped = unEscapeHtml(text);
+      const cleaned = cleanHtml(unEscaped, (level + 1) as any);
+      return {
+        id: uniqueId + i,
```

Comment:
Good point. In the end I went with componentId. It's a way to make sure you would not get duplicate id's in the html if this component is used multiple times. I'm open to suggestions of course

Reply:
Yes I get it, the rename makes sense in this case.

I did quickly google this, the chance of nanoid() being a duplicate is extremely low: https://zelark.github.io/nano-id-cc/. However, I suppose it could happen.